aten/src/ATen/native/vulkan/VulkanOps.cpp

```diff
@@ -259,12 +259,20 @@ VBuffer kernelNCHW_OCHW_repack_O4C4HWi4o4(
 
 VBuffer bufferFromOptionalHostData(
     c10::optional<const float*> data,
-    const uint32_t size) {
+    const uint32_t dataSize,
+    const uint32_t bufferSize) {
+  TORCH_INTERNAL_ASSERT(
+      dataSize <= bufferSize,
+      "buffer size(",
+      bufferSize,
+      ") is not enough for data(",
+      dataSize,
+      ")");
   const auto sizeAligned =
-      ROUND_UP(size, context().limits().minStorageBufferOffsetAlignment);
+      ROUND_UP(bufferSize, context().limits().minStorageBufferOffsetAlignment);
   VBuffer buffer{sizeAligned};
   if (data.has_value()) {
-    buffer.copy_from_host_to_device((void*)*data, size);
+    buffer.copy_from_host_to_device((void*)*data, dataSize);
```

Comment:
Any pointer type can implicity cast to void *.  The cast should not be necessary.